pennylane/tape/interfaces/autograd.py

```diff
@@ -149,17 +149,73 @@ def get_parameters(self, trainable_only=True, return_arraybox=False):
 
         return params if return_arraybox else autograd.builtins.list(params)
 
-    @autograd.extend.primitive
     def _execute(self, params, device):
-        # unwrap all NumPy scalar arrays to Python literals
-        params = [p.item() if p.shape == tuple() else p for p in params]
-        params = autograd.builtins.tuple(params)
+        """Wrapper to cath the needless autograd forward pass
+
+        This function is not primitive to autograd so when params
+        is a autograd box we know that the following call to
+        the primitve wrapped function is the autograd forward pass.
+        """
+        if isinstance(params, autograd.extend.Box):
+            self.delay_execute_device_as_this_is_a_potentially_needless_autograd_forward_pass = True

```

Comment:
I understand now that the forward pass result is needed in case the QNode is not the last computational node.